lib/azure/armrest/template_deployment_service.rb

```diff
@@ -74,6 +74,16 @@ def exists?(deploy_name, resource_group = configuration.resource_group)
 
       private
 
+      # Don't transform templates or parameters for deployments.
+      #
+      def transform_create_options(hash)
+        properties = hash['properties'] || hash[:properties]
+        ignored = properties.extract!(:template, 'template', :parameters, 'parameters') if properties
+        hash = hash.deep_transform_keys{ |k| k.to_s.camelize(:lower) }
+        (hash['properties'] || hash[:properties]).merge!(ignored) if properties
```

Comment:
may it be simplified as
```
  properties = ...
  return super(hash) unless properties
  ignored = ...
  super(hash).merge!(ignored)
```

Reply:
Eh, it's not that much simpler. Plus, the ignored hash has to be merged at the right level so your code won't pass the specs.

Reply:
@bzwei ok, I did some minor cleanup, but I think that's the best I can do for now. We can always refactor later if you think there's a better way.